hyperspy/io_plugins/emd.py

```diff
@@ -0,0 +1,392 @@
+        sample_group = emd_file.create_group('sample')
+        for key, value in self.sample.iteritems():
+            sample_group.attrs[key] = value
+        # Write comments:
+        comments_group = emd_file.create_group('comments')
+        for key, value in self.comments.iteritems():
+            comments_group.attrs[key] = value
+        # Write signals:
+        signal_group = emd_file.create_group('signals')
+        for signal in self.signals.values():
+            self._write_signal_to_group(signal_group, signal)
+        # Close file and return EMD object:
+        emd_file.close()
+
+    def print_info(self):
+        """Print all relevant information about the EMD instance.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        None
+
+        """
+        self._log.debug('Calling print_info')
+        print '\nUser:\n--------------------'
+        for key, value in self.user.iteritems():
+            print '{}:'.format(key).ljust(15), value
+        print '--------------------\n\nMicroscope:\n--------------------'
+        for key, value in self.microscope.iteritems():
+            print '{}:'.format(key).ljust(15), value
+        print '--------------------\n\nSample:\n--------------------'
+        for key, value in self.sample.iteritems():
+            print '{}:'.format(key).ljust(15), value
+        print '--------------------\n\nComments:\n--------------------'
+        for key, value in self.comments.iteritems():
+            print '{}:'.format(key).ljust(15), value
+        print '--------------------\n\nData:\n--------------------'
+        for key, value in self.signals.iteritems():
+            print '{}:'.format(key).ljust(15), value
+            print value.metadata.Signal
+        print '--------------------\n'
+
+
+def file_reader(filename, load_to_memory=True, print_info=False, **kwds):
+    emd = EMD.load_from_emd(filename, load_to_memory)
+    if print_info:
+        emd.print_info()
+    dictionaries = []
+    for signal in emd.signals.values():
+        axes = []
+        for i in range(len(signal.data.shape)):
+            axes.append({'size': signal.axes_manager[i].size,
+                         'index_in_array': signal.axes_manager[i].index_in_array,
+                         'name': signal.axes_manager[i].name,
+                         'scale': signal.axes_manager[i].scale,
+                         'offset': signal.axes_manager[i].offset,
+                         'units': signal.axes_manager[i].units})
+        dictionary = {'data': signal.data,
+                      'axes': axes,
+                      'metadata': signal.metadata.as_dictionary(),
+                      'original_metadata': signal.original_metadata.as_dictionary()}
+        dictionaries.append(dictionary)
+    return dictionaries
+
+
+def file_writer(filename, signal, signal_metadata=None, user=None,
+                microscope=None, sample=None, comments=None, **kwds):

```

Comment:
I did my best to implement this behaviour, but it feels a bit off that I have to know the exact position of the needed dictionaries in `signal.metadata` or `signal.original_metadata`. It also currently only works if the signal was loaded by my EMD-reader.
I also made sure that on load, the `metadata` are also saved as `original_metadata`, which seems a bit redundant, because they are initially completely identical. I assume this is okay, because the `metadata` can be changed during data processing, while the `original_metadata` are kept the same?
Any suggestions are highly welcome!